apps/teamsjs-diagnostic-app/src/utils/HandleRunScenario.tsx

```diff
@@ -0,0 +1,673 @@
+import { AdaptiveCardDialogInfo, barCode, profile, stageView } from '@microsoft/teams-js';
+import {
+  appInstallDialog_CheckAppInstallCapability,
+  appInstallDialog_OpenAppInstallDialog,
+  AppInstallDialogInput
+} from '../apis/AppInstallDialogApi';
+import {
+  barCode_CheckBarCodeCapability,
+  barCode_HasBarCodePermission,
+  barCode_ScanBarCode,
+} from '../apis/BarCodeApi';
+import {
+  calendar_CheckCalendarCapability,
+  calendar_ComposeMeeting,
+  calendar_OpenCalendarItem
+} from '../apis/CalendarApi';
+import {
+  call_CheckCallCapability,
+  call_StartCall
+} from '../apis/CallApi';
+import {
+  chat_CheckChatCapability,
+  chat_CloseConversation,
+  chat_OpenChat,
+  chat_OpenConversation,
+  chat_OpenGroupChat
+} from '../apis/ChatApi';
+import { dialog_CheckDialogCapability } from '../apis/DialogApi';
+import { ApiComponent } from '../components/sample/ApiComponents';
+import { dialogCard_CheckDialogAdaptiveCardCapability, dialogCard_OpenAdaptiveCardDialog } from '../apis/DialogCardApi';
+import { pages_CheckCapability, pages_GetConfig, pages_NavigateCrossDomain, pages_NavigateToApp, pages_RegisterFocusEnterHandler, pages_RegisterFullScreenChangeHandler, pages_SetCurrentFrame, pages_ShareDeepLink } from '../apis/PagesApi';
+import { profile_CheckProfileCapability, profile_ShowProfile } from '../apis/ProfileApi';
+import { search_CloseSearch, search_RegisterHandlers } from '../apis/SearchApi';
+import { clipboard_CheckClipboardCapability, clipboard_CopyImage, clipboard_CopyText, clipboard_Paste } from '../apis/ClipboardApi';
+import { geolocation_CheckGeoLocationCapability, geolocation_CheckGeoLocationMapCapability, geolocation_ChooseLocation, geolocation_GetCurrentLocation, geolocation_HasGeoLocationPermission, geolocation_RequestGeoLocationPermission } from '../apis/GeolocationApi';
+import { sharing_CheckSharingCapability, sharing_ShareWebContent } from '../apis/SharingApi';
+import { stageView_CheckStageViewCapability, stageView_OpenStageView } from '../apis/StageViewApi';
+import { people_CheckPeopleCapability, people_SelectPeople } from '../apis/PeopleApi';
+import { menus_CheckMenusCapability, menus_SetNavBarMenu, menus_SetUpViews, menus_ShowActionMenu } from '../apis/MenusApi';
+import { pagesTabs_CheckPagesTabsCapability, pagesTabs_GetMruTabInstances, pagesTabs_GetTabInstances, pagesTabs_NavigateToTab } from '../apis/PagesTabsApi';
+import { teamsCore_CheckTeamsCoreCapability, teamsCore_EnablePrintCapability, teamsCore_Print, teamsCore_RegisterBeforeUnloadHandler, teamsCore_RegisterOnLoadHandler } from '../apis/TeamsCoreApi';
+import { secondaryBrowser_CheckSecondaryBrowserCapability, secondaryBrowser_Open } from '../apis';
+
+export const handleRunScenario = async (api: ApiComponent, func: string, input?: string) => {
+  try {
+    let result;
+
+    switch (api.name) {
```

Comment:
I would not suggest structuring code so that it is a long switch statement made up of smaller switch statements. If you forget a break statement somewhere, the likelihood of making this very difficult to read and update goes very high very quickly. I would at least take each inner switch statement and make them helper functions to make this easier to read and maintain.

Reply:
Fixed this!

Reply:
Fixed it!